frontend/stmt_inferrer.py

```diff
@@ -218,19 +188,11 @@ def _infer_control_flow(node, context):
     """
     body_type = _infer_body(node.body, context)
     else_type = _infer_body(node.orelse, context)
+    result_type = z3_types.new_z3_const("control_flow")
```

Comment:
Why is there no inference for the test expression of the If/While statement?

Because of this, the inference for this program does not infer the right thing for L2 and j:

```
for x in [1, 2, 3]:

    L1 = [5, 6, 7, 8]
    L2 = [9, 10, 7, 12]
    for i in [0, 1, 2, 3]:
        for j in [0, 1, 2, 3]:
            if L1[i] == L2[j]:
                n = L1[i]
```

We need tests.